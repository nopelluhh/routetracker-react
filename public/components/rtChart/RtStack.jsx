import React, { Component } from 'react'
import * as d3 from 'd3'

class RtStack extends Component {
    componentDidMount() {
        makeChart.apply(this)
    }

    render() {
        return (
            <div style={{width: '100px', height: '100px'}} ref={ (el) => this.el = el }></div>
        )
    }

    
}

function makeChart() {
    if (!this.props.data) return
    
    const { data, width, height = '100'} = this.props

    const total = data.reduce((acc, cur) => acc + cur.value, 0)

    const chart = d3
        .select(this.el)
        .append('svg')
        .attr('viewBox', '0 0 100 100')
        .attr('width', width)
        .attr('height', height)

    const x = d3.scaleLinear()
        .domain([0, total])
        .range([0, 100])

    const segments = chart.selectAll('g')
        .data(data)
        .enter()
        .append('g')
        .attr('transform', function(d) {
            return 'translate(' + x(d) + ', 0)'
        })
    
    segments.append('rect')
        .attr('width', (d) => x(d.value))
        .attr('height', height)
        .attr('x', (d) => x(d.value))
        .attr('fill', 'black')

}

export default RtStack