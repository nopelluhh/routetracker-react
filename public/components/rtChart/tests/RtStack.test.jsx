import RtStack from '../RtStack'

import React from 'react'
import renderer from 'react-test-renderer'

describe('RtStacked tests', () => {

	const data = [{
		value: 15
	}, {
		value: 10
	}, {
		value: 50
	}]

	it('should adjust style based on props', () => {
		const component = renderer.create(
			<RtStack data={data}></RtStack>
		)

		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
