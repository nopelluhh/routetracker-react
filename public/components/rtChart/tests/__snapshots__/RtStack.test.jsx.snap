// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RtStacked tests should adjust style based on props 1`] = `
<div
  style={
    Object {
      "height": "100px",
      "width": "100px",
    }
  }
/>
`;
